src/07-embedding/03-artworks.ts

```diff
@@ -0,0 +1,278 @@
+import { ARTWORK_IDS } from "./artwork-ids"
+import weaviate from "weaviate-ts-client"
+import _ from "lodash"
+
+const client = weaviate.client({
+  scheme: "https",
+  host: "https://weaviate.stg.artsy.systems",
+})
+
+const CLASS_NAME: string = "SmallNewTrendingArtworks"
+
+const BATCH_SIZE = 100
+
+async function main() {
+  const artworks = await fetchArtworks()
+
+  console.log(`Fetched ${artworks.length} artworks`)
+  console.log(artworks)
+  await prepareArtworksCollection()
+  await insertArtworks(artworks)
+  /*
+   * now query from Weaviate console:
+   * https://console.weaviate.io
+   */
+}
+
+main()
+
+async function prepareArtworksCollection() {
+  const client = weaviate.client({
+    scheme: "https",
+    host: "https://weaviate.stg.artsy.systems",
+  })
+
+  const alreadyExists = await client.schema.exists(CLASS_NAME)
+
+  if (alreadyExists) {
+    console.log(`${CLASS_NAME} class already exists, deleting it`)
+    await client.schema.classDeleter().withClassName(CLASS_NAME).do()
+  }
+
+  const classWithProps = {
+    class: CLASS_NAME,
+    vectorizer: "text2vec-openai",
+    moduleConfig: {
+      "text2vec-openai": {
+        model: "text-embedding-3-small",
+        dimensions: 1536,
+        type: "text",
+        vectorizeClassName: false,
+      },
+      "generative-openai": {
+        model: "gpt-3.5-turbo",
+      },
+    },
+    properties: [
+      {
+        name: "rarity",
+        dataType: ["text"],
```

Comment:
With the version bump of `weaviate` from `1.18.0` to `1.24.14`, we could now use the `object` `dataType`, which was [added](https://weaviate.io/developers/weaviate/config-refs/datatypes#datatype-object) in `1.22`. 

However, it seemed to make sense to flatten the data structure in this case to match the other experiments. If we were to use `weaviate` in any kind of production capacity, I think it would make more sense to keep the schema closer to MP, to reduce schema variations between systems.